core/unix/signal_linux_x86.c

```diff
@@ -468,8 +515,37 @@ sigcontext_to_mcontext_simd(priv_mcontext_t *mc, sig_full_cxt_t *sc_full)
                 }
             }
         }
+#ifdef X64
+        if (ZMM_ENABLED()) {
+            kernel_xstate_t *xstate = (kernel_xstate_t *)sc->fpstate;
+            if (sc->fpstate->sw_reserved.magic1 == FP_XSTATE_MAGIC1) {
+                ASSERT(TEST(XCR0_ZMM_HI256, sc->fpstate->sw_reserved.xstate_bv));
+                ASSERT(TEST(XCR0_HI16_ZMM, sc->fpstate->sw_reserved.xstate_bv));
```

Comment:
Why would both of these have to be set?  Does that always follow from ZMM_ENABLED?  Same for the opmask below.  These are all separate cpuid bits so it seems possible to not have all of them.

Reply:
That's because Intel requires all states to be enabled in order to determine if AVX-512 is enabled. See15.2 DETECTION OF AVX-512 FOUNDATION INSTRUCTIONS. That's actually how we determine ZMM_ENABLED(). Therefore the ASSERTs. Resolving, please re-open if necessary.

Reply:
Requesting a comment since this was not clear.  The placement of the asserts made me think they were related to FP_XSTATE_MAGIC1 or sthg.  Maybe move all 3 up just inside ZMM_ENABLED, or put a comment next to where they are now.

Reply:
Comment added, done.